### PR TITLE
[MKC-1351] OPAMP Docs

### PR DESCRIPTION
**Pull request by Hannes7eicher**
## What This PR Changes
This PR adds documentation on how to use the OPAMP on the Arduino UNO R4 Minima and R4 WiFi.

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/assets/circuitAmplifier.png
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/assets/circuitAmplifier.png:1]: The text is really hard to read, can we make it bigger?

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/assets/circuitFollower.png
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/assets/circuitFollower.png:1]: same with this text.

Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/assets/circuitFollower.png:1]: The analog pins have the wrong names here. you've marked A0 as A1

Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/assets/circuitFollower.png:1]: Also what happened to the illustration? There are caps everywhere, even on top of the RA4M1

File: content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/assets/circuitAmplifier.png
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/assets/circuitAmplifier.png:1]: Same comments as the WiFi version

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+- about the OPAMP feature onboard the UNO R4 WiFi,
+- about the basics of the OPAMP library, 
+- how to mirror a voltage signal,
+- how to amplify a voltage signal.
+
+## Hardware & Software Needed
+  To follow along with this article, you will need the following hardware: 
+  
+  - [Arduino UNO R4 WiFi](https://store.arduino.cc/uno-r4-wifi)
+  - resistors
```
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md:29]: What value?

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+An OPAMP is a versatile and widely used electronic component that belongs to the class of analog integrated circuits. Its primary function is to amplify voltage signals but they are very versatile and can be used to:
+
+- mirror an input voltage to its output,
+- amplify a small analog voltage to its output pin, output voltage range from 0 to ~4.7 V,
+- compare two input voltages and give a binary "higher" or "lower" output,
+- integrate and differentiate signals.
+
+## Voltage Follower
+
+The most simple way of testing the OPAMP is to create a voltage follower, by connecting `A2` to `A3`. Then, any voltage applied to `A1` should be mirrored on `A3`. For example, if you connect `A1` to `GND`, the OPAMP output should be `GND`. Connect `A1` to `3.3 V`, the output should be `3.3 V`.
```
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md:44]:
```suggestion
The simplest way to test the OPAMP is to configure it as a voltage follower by connecting A2 to A3. In this setup, the voltage at A3 should mimic the voltage applied to A1. For instance, if you connect A1 to the ground (GND), the OPAMP output at A3 should also be at the ground potential. Similarly, if you connect A1 to 3.3 V, the output at A3 should be approximately 3.3 V.
```

**Review by jacobhylen — CHANGES_REQUESTED** [review]
There are a couple of things that need changing before we go live, but it's a good start.

Apart from the other comments I left, there are a few things I feel are missing.
- This shows how to set it up for a 2x amplification, but where is this configured? Can i amplify by another factor?
- comparator functionality, we say it can be used for this but never show how
- If I follow the article, there aren't really any loose ends to grasp at, I am left with an incomplete project and am asking myself what to explore next?

**Comment by jacobhylen** [comment]
Also we should warn not to amplify a battery, as you'd essentially short circuit it. Depending on the battery this could be dangerous.

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md (hunk 1 above)
Comment (Hannes7eicher) [reply content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md:29]: Depends on the amplification. I think the section explaining how to calculate the resistor value should be helpful.

File: content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  - resistors 
+  - Jumper wires
+  
+  
+## Operational Amplifier (OPAMP)
+
+An OPAMP is a versatile and widely used electronic component that belongs to the class of analog integrated circuits. Its primary function is to amplify voltage signals but they are very versatile and can be used to:
+
+- mirror an input voltage to its output,
+- amplify a small analog voltage to its output pin, output voltage range from 0 to ~4.7 V,
```
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md:38]:
```suggestion
- amplify a small analog voltage to its output pin, the UNO R4's output voltage range is 0 to ~4.7 V,
```

Comment (jacobhylen) [reply content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md:38]: Makes it clear in the section about generic OPAMPS that this note is about the UNO R4 OPAMP specifically :-)

File: content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+The simplest way to test the OPAMP is to configure it as a voltage follower by connecting A2 to A3. In this setup, the voltage at A3 should mimic the voltage applied to A1. For instance, if you connect A1 to the ground (GND), the OPAMP output at A3 should also be at the ground potential. Similarly, if you connect A1 to 3.3 V, the output at A3 should be approximately 3.3 V.
+
+## Circuit
+
+![Voltage Follower Circuit](./assets/circuitFollowerMinima.png)
+
+## Voltage Amplifier
+
+***Do NOT amplify the voltage provided by a battery as this can pose serious safety risks!***
```
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md:52]:
```suggestion
***Caution: When amplifying the voltage of a battery using an operational amplifier, you should be aware of potential risks. The amplification process can lead to a high current draw, essentially placing a significant load on the battery, which may result in overheating, damage, or even pose a security risk. To mitigate this risk, you may for example add a resistor in series with the battery.***
```

File: content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`.
```
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md:54]:
```suggestion
A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. a circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`. 
```

**Review by jacobhylen — COMMENTED** [review]
Much better shape, I left a few more comments (sorry, nitpicking), and then we're good to go.

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  - resistors
+  - Jumper wires
+  
+  
+## Operational Amplifier (OPAMP)
+
+An OPAMP is a versatile and widely used electronic component that belongs to the class of analog integrated circuits. Its primary function is to amplify voltage signals but they are very versatile and can be used to:
+
+- mirror an input voltage to its output,
+- amplify a small analog voltage to its output pin, output voltage range from 0 to ~4.7 V,
```
Comment (Hannes7eicher) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md:38]:
```suggestion
- amplify a small analog voltage to its output pin, the UNO R4's output voltage range is 0 to ~4.7 V,
```

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+The simplest way to test the OPAMP is to configure it as a voltage follower by connecting A2 to A3. In this setup, the voltage at A3 should mimic the voltage applied to A1. For instance, if you connect A1 to the ground (GND), the OPAMP output at A3 should also be at the ground potential. Similarly, if you connect A1 to 3.3 V, the output at A3 should be approximately 3.3 V.
+
+## Circuit
+
+![Voltage Follower Circuit](./assets/circuitFollowerWiFi.png)
+
+## Voltage Amplifier
+
+***Do NOT amplify the voltage provided by a battery as this can pose serious safety risks!***
```
Comment (Hannes7eicher) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md:52]:
```suggestion
***Caution: When amplifying the voltage of a battery using an operational amplifier, you should be aware of potential risks. The amplification process can lead to a high current draw, essentially placing a significant load on the battery, which may result in overheating, damage, or even pose a security risk. To mitigate this risk, you may for example add a resistor in series with the battery.***
```

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`.
```
Comment (Hannes7eicher) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md:54]:
```suggestion
A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. a circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`. 
```

File: content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md (hunk 4 above)
Comment (Hannes7eicher) [reply content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md]: Really good explanation!

File: content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+## Circuit
+
+![Voltage Follower Circuit](./assets/circuitFollowerMinima.png)
+
+## Voltage Amplifier
+
+***Caution: When amplifying the voltage of a battery using an operational amplifier, you should be aware of potential risks. The amplification process can lead to a high current draw, essentially placing a significant load on the battery, which may result in overheating, damage, or even pose a security risk. To mitigate this risk, you may for example add a resistor in series with the battery.***
+
+A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. a circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`.
```
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-minima/tutorials/opamp/opamp.md:54]:
```suggestion
A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. A circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`. 
```

File: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+## Circuit
+
+![Voltage Follower Circuit](./assets/circuitFollowerWiFi.png)
+
+## Voltage Amplifier
+
+***Caution: When amplifying the voltage of a battery using an operational amplifier, you should be aware of potential risks. The amplification process can lead to a high current draw, essentially placing a significant load on the battery, which may result in overheating, damage, or even pose a security risk. To mitigate this risk, you may for example add a resistor in series with the battery.***
+
+A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. a circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`.
```
Comment (jacobhylen) [inline content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md:54]:
```suggestion
A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. A circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`. 
```

**Review by jacobhylen — APPROVED** [review]
LGTM!

**Pull request merged**